### PR TITLE
add screen airnow hourly and daily obs for AQM and GEFS

### PR DESCRIPTION
**Pull request by Ho-ChunHuang-NOAA**
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

Screen airnow hourly and daily ascii file for inconsistent column.  If the column number of first 3 records of daily airnow are different, use the default defined in ~/job/JEVS_AQM_PREP.

This fix impacts both AQM and GEFS-CHEM grib2obs.

## Developer Questions and Checklist
* Is this a high priorty PR? If so, why and is there a date it needs to be merged by?
NO

* Do you have any planned upcoming annual leave/PTO?
NO

* Are there any changes needed for when the jobs are supposed to run?
NO
  
- [ ] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
YES

- [ ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
YES

- [ ] References the feature branch for `HOMEevs` are removed from the code.
YES

- [ ] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
YES

- [ ] Jobs over 15 minutes in runtime have restart capability.
NA

- [ ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
NA

- [ ] Jobs contain the approriate file checking and don't run METplus for any missing data.
YES

- [ ] Code is using METplus wrappers structure and not calling MET executables directly.
YES

- [ ] Log is free of any ERRORs or WARNINGs.
YES

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

Run today's prep for both AQM and GEFS to test free of error in normal condition.

Run AQM and GEFS prep for
INITDATE=20241015 and
export DCOMINairnow=/lfs/h2/emc/vpppg/noscrub/ho-chun.huang/dcom/prod/airnow

Look for the message output for vldhr 02, 07, and 10. for AQM and GEFS output, and daily airnow for AQM log output

vdlhr=02
DEBUG :: Line 6 has different columns number 23 vs standard 34
DEBUG :: Line 8 has different columns number 39 vs standard 34

vldhr=07
DEBUG :: Line 2 has different columns number 21 vs standard 34
DEBUG :: Line 3 has different columns number 14 vs standard 34
DEBUG :: Line 4 has different columns number 46 vs standard 34
DEBUG :: Line 5 has different columns number 46 vs standard 34

vldhr=10
DEBUG :: Line 2 has different columns number 26 vs standard 34
DEBUG :: Line 3 has different columns number 23 vs standard 34
DEBUG :: Line 4 has different columns number 133 vs standard 34

daily airnow
DEBUG :: Line 2 has different columns number 7 vs standard 13

### CODE REVIEW

File: scripts/prep/aqm/exevs_aqm_prep.sh
Comment (malloryprow) [inline scripts/prep/aqm/exevs_aqm_prep.sh:74]: I believe there should be a check here for `$screen_file` before running METplus.

File: scripts/prep/aqm/exevs_aqm_prep.sh
Comment (malloryprow) [inline scripts/prep/aqm/exevs_aqm_prep.sh:118]: I believe there should be a check here for `$screen_file` before running METplus.

File: scripts/prep/global_ens/exevs_global_ens_gefs_chem_grid2obs_prep.sh
Comment (malloryprow) [inline scripts/prep/global_ens/exevs_global_ens_gefs_chem_grid2obs_prep.sh:110]: I believe there should be a check here for `$screen_file` before running METplus.

**Comment by Ho-ChunHuang-NOAA** [comment]
Do you mean adding

export err=$?; err_chk

after each python ${USHevs}/${COMPONENT}/screen_airnow_obs_*.py ?

**Comment by malloryprow** [comment]
That would be good to add too!

I think there should be a file check for `$screen_file` before METplus is run. We don't want to run METplus if the file doesn't exist because that will cause an error from METplus. If I'm following correctly, `$screen_file` is what is being used as input for ASCII2NC now.

**Comment by Ho-ChunHuang-NOAA** [comment]
screen_file is the output of previous python execution

**Comment by Ho-ChunHuang-NOAA** [comment]
if the python screen code works it will exist in the {DATA} directory, as well as if it is not existed "number_of_record=$()" will failed.

**Comment by malloryprow** [comment]
Oh yes, I see that now. Thanks!

Could we get those `export err=$?; err_chk` after the python script executions?

**Comment by Ho-ChunHuang-NOAA** [comment]
add err_chk

**Comment by malloryprow** [comment]
### Today's run

### jevs_aqm_prep.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/EVS/dev/drivers/scripts/prep/aqm/jevs_aqm_prep_00.o159301186
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_aqm_prep_00.159301186.cbqs01

### jevs_global_ens_chem_grid2obs_prep.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/EVS/dev/drivers/scripts/prep/global_ens/jevs_global_ens_chem_grid2obs_prep.o159301254
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_global_ens_chem_grid2obs_prep.159301254.cbqs01

**Comment by Ho-ChunHuang-NOAA** [comment]
The logs file look good to me.

AQM prep output size is the same, but daily netcdf file is slightly different.
emc_parallel ncdump 
dimensions:
        mxstr = 16 ;
        mxstr2 = 40 ;
        mxstr3 = 80 ;
        nhdr = 2104 ;
        nobs = 4893 ;
        nhdr_typ = 1 ;
        **nhdr_sid = 1678** ;
        nhdr_vld = 1 ;
        nobs_qty = 1 ;
        obs_var_num = 6 ;

pr ncdump

dimensions:
        mxstr = 16 ;
        mxstr2 = 40 ;
        mxstr3 = 80 ;
        nhdr = 2104 ;
        nobs = 4893 ;
        nhdr_typ = 1 ;
        **nhdr_sid = 2104 ;**
        nhdr_vld = 1 ;
        nobs_qty = 1 ;
        obs_var_num = 6 ;

The non-checked and checked daily airnow are identical
/lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_aqm_prep_00.159301186.cbqs01 > diff /lfs/h1/ops/prod/dcom/20241022/airnow/daily_data_v2.dat checked_daily_data_v2.dat

I think it is okay for AQM

The GEFS file size look good to me.

**Comment by malloryprow** [comment]
### INITDATE=20241015's run

### jevs_aqm_prep.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/EVS/dev/drivers/scripts/prep/aqm/jevs_aqm_prep_00.o159306908
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_aqm_prep_00.159306908.cbqs01

### jevs_global_ens_chem_grid2obs_prep.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/EVS/dev/drivers/scripts/prep/global_ens/jevs_global_ens_chem_grid2obs_prep.o159307210
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr596/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_global_ens_chem_grid2obs_prep.159307210.cbqs01

**Comment by Ho-ChunHuang-NOAA** [comment]
The logs file generated the debugged message as expected

DEBUG :: Line 6 has different columns number 23 vs standard 34
DEBUG :: Line 8 has different columns number 39 vs standard 34

DEBUG :: Line 2 has different columns number 21 vs standard 34
DEBUG :: Line 3 has different columns number 14 vs standard 34
DEBUG :: Line 4 has different columns number 46 vs standard 34
DEBUG :: Line 5 has different columns number 46 vs standard 34

DEBUG :: Line 2 has different columns number 26 vs standard 34
DEBUG :: Line 3 has different columns number 23 vs standard 34
DEBUG :: Line 4 has different columns number 133 vs standard 34

DEBUG :: Line 2 has different columns number 7 vs standard 13

**Review by AliciaBentley-NOAA — APPROVED** [review]
I have reviewed this PR and think that the changes and additions look great. I approve this PR to be merged after testing.  

CC @Ho-ChunHuang-NOAA @malloryprow @AndrewBenjamin-NOAA

**Review by malloryprow — APPROVED** [review]
Changes are good and testing successful.

**Comment by malloryprow** [comment]
We have 2 approvals but will wait for @AndrewBenjamin-NOAA to merge since AQM is regional!

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
I concur with Alicia's comments and also approve this PR. Thank you for testing this across both the regional and global AQ components.

**Comment by AndrewBenjamin-NOAA** [comment]
> We have 2 approvals but will wait for @AndrewBenjamin-NOAA to merge since AQM is regional!

You should already have it! I just approved a few minutes ago

**Pull request merged**
